server/channels/app/post_metadata.go

```diff
@@ -362,6 +362,11 @@ func (a *App) getImagesForPost(c request.CTX, post *model.Post, imageURLs []stri
 	}
 
 	for _, imageURL := range imageURLs {
+		// MM-51060 - don't accept Mattermost permalinks as Open Graph Image URLs.
+		if isPermalink := looksLikeAPermalink(imageURL, a.GetSiteURL()); isPermalink {
```

Comment:
I was going to test this, but since spinwicks are apparently broken, I'll just ask: Here we're doing `looksLikeAPermalink` with the original image URL, but later, before the next call to `looksLikeAPermalink` in `getLinkMetadata`, the URL goes through ` resolveMetadataURL` first.

Is that an issue? Seems like there's a chance that the two URLs actually end up being different, and `looksLikeAPermalink` might return false here but true later?